src/app/ngrx/effects/castle.effects.ts

```diff
@@ -1,54 +1,231 @@
+      return this.castle.followItems('prx:podcasts', { page, per: PODCAST_PAGE_SIZE })
+      .pipe(
+        map((results: HalDoc[]) => {
+          if (!results.length) {
+            const error = 'Looks like you don\'t have any podcasts.';
+            return new ACTIONS.CastlePodcastPageFailureAction({error});
+          } else {
+            return new ACTIONS.CastlePodcastPageSuccessAction({
+              page,
+              all,
+              total: results[0].total(),
+              podcasts: results.map(doc => {
+                return {
+                  id: '' + doc['id'],
+                  title: doc['title']
+                };
+              })
+            });
+          }
+        }),
+        catchError(error => Observable.of(new ACTIONS.CastlePodcastPageFailureAction({error})))
+      );
+    })
+  );
+
+  // on podcast page success if not already routed to a podcast,
+  // check for podcast id on local storage or use first podcast id in list and route to it
+  @Effect()
+  loadPodcastsSuccess$: Observable<Action> = this.actions$.pipe(
+    ofType(ACTIONS.ActionTypes.CASTLE_PODCAST_PAGE_SUCCESS),
+    // only dispatches a routing action when there is not already a routed :podcastId
+    filter(() => !this.routerParams.podcastId),
+    map((action: ACTIONS.CastlePodcastPageSuccessAction) => action.payload),
+    switchMap((payload: ACTIONS.CastlePodcastPageSuccessPayload) => {
+      const { podcasts } = payload;
+      const localStorageRouterParams: RouterParams = localStorageUtil.getItem(localStorageUtil.KEY_ROUTER_PARAMS);
+      const localStoragePodcastInList = localStorageRouterParams && localStorageRouterParams.podcastId &&
+        podcasts.find(podcast => podcast.id === localStorageRouterParams.podcastId);
+      return Observable.of(new ACTIONS.RoutePodcastAction( {
+        // navigate to either the podcastStorageId in localStorage or the first one in the result from CMS (which is the last one changed)
+        podcastId: (localStoragePodcastInList && localStorageRouterParams.podcastId) || podcasts[0].id
+      }));
+    })
+  );
+
+  // on podcast page success if loading all podcast pages and not yet finished, load next page
+  @Effect({dispatch: false})
+  loadNextPodcastPage$: Observable<Action> = this.actions$.pipe(
+    ofType(ACTIONS.ActionTypes.CASTLE_PODCAST_PAGE_SUCCESS),
+    filter((action: ACTIONS.CastlePodcastPageSuccessAction) => {
+      const { page, all, total } = action.payload;
+      return all && page * PODCAST_PAGE_SIZE < total;
+    }),
+    map((action: ACTIONS.CastlePodcastPageSuccessAction) => action.payload),
+    concatMap((payload: ACTIONS.CastlePodcastPageSuccessPayload) => {
+      const { page, all } = payload;
+      return Observable.of(new ACTIONS.CastlePodcastPageLoadAction({page: page + 1, all}));
+    })
+  );
+
+  /*
+  // on podcast page 1 success, if loading all podcasts start loading all episodes page by page as well
+  // ugh, makes app loading too slow
```

Comment:
I could probably get rid of this commented out effect, but tbh I'm still debating how to load all the episodes to produce the new menu for selecting a single episode. There is an `/episodes` endpoint I might try next instead of loading each `/podcast/:podcastId/episodes` 
<img width="633" alt="screen shot 2018-07-31 at 11 56 46 am" src="https://user-images.githubusercontent.com/2250413/43976686-20b255a4-9ca7-11e8-92cc-d2ef86b266cb.png">